include/dlaf/factorization/cholesky/mc/cholesky_L.h

```diff
@@ -156,38 +156,35 @@ void cholesky_L(comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_a) {
         // Avoid useless communications if one-column communicator and if on the last column
         if (col_comm_size > 1 && k != (mat_a.nrTiles().cols() - 1)) {
           // Receive the diagonal tile
-          kk_tile = hpx::dataflow(  //
-              executor_mpi,
-              hpx::util::unwrapping(
-                  [](auto index, auto&& tile_size, auto&& comm_wrapper) -> Tile<const T, Device::CPU> {
-                    memory::MemoryView<T, Device::CPU> mem_view(
-                        util::size_t::mul(tile_size.rows(), tile_size.cols()));
-                    Tile<T, Device::CPU> tile(tile_size, std::move(mem_view), tile_size.rows());
-                    comm::sync::broadcast::receive_from(index, comm_wrapper().colCommunicator(), tile);
-                    return std::move(tile);
-                  }),
-              k_rank_row, mat_a.tileSize(GlobalTileIndex(k, k)), serial_comm());
+          auto recv_bcast_f = unwrapping([rank = kk_tile_rank.row(), tile_size = mat_a.tileSize(kk_idx)](
+                                             auto&& comm_wrapper) -> ConstTile_t {
+            MemView_t mem_view(util::size_t::mul(tile_size.rows(), tile_size.cols()));
+            Tile_t tile(tile_size, std::move(mem_view), tile_size.rows());
+            comm::sync::broadcast::receive_from(rank, comm_wrapper().colCommunicator(), tile);
+            return std::move(tile);
+          });
+          kk_tile = hpx::dataflow(executor_mpi, std::move(recv_bcast_f), serial_comm());
         }
       }
 
+      auto k_local_col = distr.localTileFromGlobalTile<Coord::Col>(k);
```

Comment:
Isn't this the same as `kk.col()`?

Reply:
`kk` is out of scope here.

Reply:
There are more improvements that can be made, perhaps we should leave them for a separate PR?